include/macros.h

```diff
@@ -1,2 +1,11 @@
+#define SQ(x) ((x) * (x))
 #define ABS(x) ((x) >= 0 ? (x) : -(x))
-#define ABS_ALT(x) ((x) < 0 ? -(x) : (x))
\ No newline at end of file
+#define ABS_ALT(x) ((x) < 0 ? -(x) : (x))
+#define DECR(x) ((x) == 0 ? 0 : --(x))
+
+#define CLAMP(x, min, max) ((x) < (min) ? (min) : (x) > (max) ? (max) : (x))
+#define CLAMP_MAX(x, max) ((x) > (max) ? (max) : (x))
+#define CLAMP_MIN(x, min) ((x) < (min) ? (min) : (x))
+
+#define ARRAY_COUNT(arr) (s32)(sizeof(arr) / sizeof(arr[0]))
```

Comment:
I find the name `ARRAY_SIZE` as slightly more appropriate

Reply:
I'm sorry but i don't agree, the MACRO "counts" how many "whatevers" are in the array based on it's size